build_test.go

```diff
@@ -478,6 +478,95 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 		})
 	})
 
+	context("when BP_NGINX_CONF_LOCATION is set to a relative path", func() {
+		it.Before(func() {
+			os.Setenv("BP_NGINX_CONF_LOCATION", "some-relative-path/nginx.conf")
+		})
+
+		it.After(func() {
+			os.Unsetenv("BP_NGINX_CONF_LOCATION")
+		})
+		it("assumes path is relative to /workspace", func() {
+			result, err := build(packit.BuildContext{
+				CNBPath:    cnbPath,
+				WorkingDir: workspaceDir,
+				Stack:      "some-stack",
+				Plan: packit.BuildpackPlan{
+					Entries: []packit.BuildpackPlanEntry{
+						{
+							Name: "nginx",
+							Metadata: map[string]interface{}{
+								"version-source": "BP_NGINX_VERSION",
+								"version":        "1.19.*",
+								"launch":         true,
+							},
+						},
+					},
+				},
+				Layers: packit.Layers{Path: layersDir},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result.Launch.Processes[0].Args[len(result.Launch.Processes[0].Args)-1]).To(Equal(filepath.Join(workspaceDir, "some-relative-path", "nginx.conf")))
+		})
+	})
+
+	context("when BP_NGINX_CONF_LOCATION is set to an absolute path", func() {
+		it.Before(func() {
+			os.Setenv("BP_NGINX_CONF_LOCATION", "/some-absolute-path/nginx.conf")
+		})
+
+		it.After(func() {
+			os.Unsetenv("BP_NGINX_CONF_LOCATION")
+		})
+		it("uses the location as-is", func() {
+			result, err := build(packit.BuildContext{
+				CNBPath:    cnbPath,
+				WorkingDir: workspaceDir,
+				Stack:      "some-stack",
+				Plan: packit.BuildpackPlan{
+					Entries: []packit.BuildpackPlanEntry{
+						{
+							Name: "nginx",
+							Metadata: map[string]interface{}{
+								"version-source": "BP_NGINX_VERSION",
+								"version":        "1.19.*",
+								"launch":         true,
+							},
+						},
+					},
+				},
+				Layers: packit.Layers{Path: layersDir},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result.Launch.Processes[0].Args[len(result.Launch.Processes[0].Args)-1]).To(Equal("/some-absolute-path/nginx.conf"))
+		})
+	})
+
+	context("when BP_NGINX_CONF_LOCATION is not set", func() {
```

Comment:
Is this test adding anything beyond what the default scenario (called: `"it does a build"`) tests? I can't see any functional difference but I could certainly be missing something.

Reply:
Good point, @robdimsdale. I guess it is ultimately the same test. This test only checks that the path is constructed correctly when the custom path is relative; the other test makes other checks including this. But I think I agree that this isn't needed.